google/cloud/aiplatform/initializer.py

```diff
@@ -149,6 +149,35 @@ def project(self) -> str:
         if self._project:
             return self._project
 
+        # Project is not set. Trying to get it from the environment.
+        # See https://github.com/googleapis/python-aiplatform/issues/852
+        # See https://github.com/googleapis/google-auth-library-python/issues/924
+        # TODO: Remove when google.auth.default() learns the
+        # CLOUD_ML_PROJECT_ID env variable or Vertex AI starts setting GOOGLE_CLOUD_PROJECT env variable.
+        project_number = os.environ.get("CLOUD_ML_PROJECT_ID")
+        if project_number:
+            self._project = project_number
+            # Try to convert project number to project ID which is more readable.
+            try:
+                from googleapiclient import discovery
+
+                cloud_resource_manager_service = discovery.build(
+                    "cloudresourcemanager", "v3"
+                )
+                project_id = (
+                    cloud_resource_manager_service.projects()
+                    .get(name=f"projects/{project_number}")
+                    .execute()["projectId"]
+                )
+                self._project = project_id
```

Comment:
I'm a bit hesitant to assign to `self._project` because it's inconsistent with how we handle retrieving the default project below. I understand this is useful because it will remove the additional API calls when accessing the property in the future. Perhaps if this is broken out into a separate utility we can use `functools.cache` or `functoosl.lru_cache` to capture the mapping of project number to project ID.

Reply:
Done.